### ISSUE TITLE
Validator CLI complains with java.lang.ClassCastException (CapabilityStatement -> TerminologyCapabilities)

### ISSUE THREAD

**Issue by imadeit** [hapifhir/org.hl7.fhir.core#831]
I am using the validator_cli.jar released recently ([v5.6.47](https://github.com/hapifhir/org.hl7.fhir.core/releases/tag/5.6.47)) to validate my local resource.
Our local terminology server is only able to deal with JSON format of capability statement, so I posted another issue #830 

Then I tried to return an XML capability statement at first round of request for /metadata?_summary=true, then another exception was thrown as follows.
```
Loading
  Load FHIR v4.0 from hl7.fhir.r4.core#4.0.1 - 4575 resources (00:06.0497)
  Load hl7.terminology#3.1.0 - 4117 resources (00:01.0744)
  Terminology server http://localhost:8081Exception in thread "main" org.hl7.fhir.exceptions.FHIRException: Unable to connect to terminology server. Use parameter '-tx n/a' to run without using terminology services to validate LOINC, SNOMED, ICD-X etc. Error = Error fetching the server's terminology capabilities
	at org.hl7.fhir.r5.context.SimpleWorkerContext.connectToTSServer(SimpleWorkerContext.java:343)
	at org.hl7.fhir.validation.ValidationEngine.connectToTSServer(ValidationEngine.java:373)
	at org.hl7.fhir.validation.ValidationEngine.connectToTSServer(ValidationEngine.java:362)
	at org.hl7.fhir.validation.ValidationEngine.setTerminologyServer(ValidationEngine.java:783)
	at org.hl7.fhir.validation.cli.services.ValidationService.initializeValidator(ValidationService.java:333)
	at org.hl7.fhir.validation.cli.services.ValidationService.initializeValidator(ValidationService.java:314)
	at org.hl7.fhir.validation.ValidatorCli.doValidation(ValidatorCli.java:241)
	at org.hl7.fhir.validation.ValidatorCli.main(ValidatorCli.java:165)
Caused by: org.hl7.fhir.exceptions.FHIRException: Error fetching the server's terminology capabilities
	at org.hl7.fhir.r5.utils.client.FHIRToolingClient.getTerminologyCapabilities(FHIRToolingClient.java:144)
	at org.hl7.fhir.convertors.txClient.TerminologyClientR5.getTerminologyCapabilities(TerminologyClientR5.java:84)
	at org.hl7.fhir.r5.context.SimpleWorkerContext.connectToTSServer(SimpleWorkerContext.java:337)
	... 7 more
Caused by: java.lang.ClassCastException: class org.hl7.fhir.r5.model.CapabilityStatement cannot be cast to class org.hl7.fhir.r5.model.TerminologyCapabilities (org.hl7.fhir.r5.model.CapabilityStatement and org.hl7.fhir.r5.model.TerminologyCapabilities are in unnamed module of loader 'app')
	at org.hl7.fhir.r5.utils.client.FHIRToolingClient.getTerminologyCapabilities(FHIRToolingClient.java:142)
	... 9 more
```
Even I return the XML downloaded from http://tx.fhir.org/r4/metadata?_summary=true, the exception is also thrown.

Is there any advice?
Thanks in advance.

**Comment by grahamegrieve** [comment]
you are not supporting the mode parameter - see http://hl7.org/fhir/http.html#capabilities

**Comment by imadeit** [comment]
@DarthGizka Thanks for kindly responding.
Here is the request and response when I am using validator-cli.jar
1st round: GET /metadata?_summary=true
https://gist.github.com/imadeit/e95741802ee8b0fcfd48afe787ada7fa
2nd round: GET /metadata?mode=terminology
https://gist.github.com/imadeit/26ffcca8820c697ac91145853cd0c727

The exceptions are as I posted days before.
As I mentioned since our Terminology server only supports application/fhir+json, I modified the source code as follows and use the built version of validator-cli. The ClassCastException exception is the same.
```
   public FHIRToolingClient(String baseServiceUrl, String userAgent) throws URISyntaxException {
-    preferredResourceFormat = ResourceFormat.RESOURCE_XML;
+    preferredResourceFormat = ResourceFormat.RESOURCE_JSON;
```
Pls. kindly confirm and thanks in advance.
